xdsl/dialects/riscv.py

```diff
@@ -331,6 +334,73 @@ class RISCVOp(Operation, ABC):
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        operands = cls.parse_operands(parser)
+        successors = cls.parse_successors(parser)
+        regions = cls.parse_regions(parser)
+        attributes = cls.parse_attributes(parser)
+        parser.parse_punctuation(":")
+        result_types = cls.parse_function_type(parser).outputs.data
+        return cls.create(
+            operands=operands,
+            result_types=result_types,
+            attributes=attributes,
+            successors=successors,
+            regions=regions,
+        )
+
+    @classmethod
+    def parse_operands(cls, parser: Parser) -> Sequence[SSAValue]:
+        return parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN, parser.parse_operand
+        )
+
+    @classmethod
+    def parse_successors(cls, parser: Parser) -> Sequence[Block]:
+        return parser.parse_optional_successors() or []
+
+    @classmethod
+    def parse_function_type(cls, parser: Parser) -> FunctionType:
+        return parser.parse_function_type()
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        return parser.parse_optional_attr_dict()
+
+    @classmethod
+    def parse_regions(cls, parser: Parser) -> Sequence[Region]:
+        if parser.parse_optional_punctuation("(") is not None:
+            regions = parser.parse_comma_separated_list(
+                parser.Delimiter.NONE, parser.parse_region
+            )
+            parser.parse_punctuation(")")
+            return regions
+        return []
+
+    def print(self, printer: Printer) -> None:
+        self.print_operands(printer)
+        printer.print_successors(self.successors)
+        printer.print_regions(self.regions)
+        self.print_attributes(printer)
+        printer.print(" : ")
+        self.print_types(printer)
+
+    def print_operands(self, printer: Printer) -> bool:
+        printer.print("(")
+        if self.operands:
+            printer.print_list(self.operands, printer.print_ssa_value)
+        printer.print(")")
+        return bool(self.operands)
+
+    def print_attributes(self, printer: Printer) -> Set[str]:
+        if self.attributes:
+            printer.print_op_attributes(self.attributes)
+        return self.attributes.keys()
+
+    def print_types(self, printer: Printer) -> None:
+        printer.print_operation_type(self)
```

Comment:
This one should also just be replaced with a call to the printer method as far as I can tell.

Reply:
I agree.

Reply:
Commented above.